File: client/src/main/js/components/reports/DeadlineReport.js
import React from 'react';

export function DeadlineReport(props) {
    const lastCommit = props.solution.commits.last();
    const lastCommitDate = lastCommit ? lastCommit.date : null;
    if (lastCommitDate) {
        if (props.task.deadline) {
            if (props.task.deadline > lastCommitDate) {
                return (
                    <section className="report valid-deadline-report">
                        <i className="material-icons">alarm_on</i>
                    </section>
                );
            } else {
                return (
                    <section className="report invalid-deadline-report">
                        <i className="material-icons">alarm_off</i>
                    </section>
                );
            }
        } else {
            return (
                <section className="report valid-deadline-report">
                    <i className="material-icons">alarm_on</i>
                </section>
            );
        }
    } else {
        return (
            <section className="report"/>
        )
    }
}

File: frontend/src/main/kotlin/org/flaxo/frontend/component/report/DeadlineReport.kt
package org.flaxo.frontend.component.report

import org.flaxo.common.data.Solution
import org.flaxo.common.data.Task
import react.RBuilder
import react.dom.i

fun RBuilder.deadlineReport(task: Task, solution: Solution) =
        solution.commits
                .lastOrNull()
                ?.date
                ?.also { latestCommitDate ->
                    val deadline = task.deadline
                    if (deadline == null || deadline > latestCommitDate) {
                        i(classes = "material-icons valid-deadline-report") { +"alarm_on" }
                    } else {
                        i(classes = "material-icons invalid-deadline-report") { +"alarm_off" }
                    }
                }
                ?: i { }
